### PR TITLE
CMSIS CORE: Add support for __SXTB16_RORn

### PR DESCRIPTION
**Pull request by gmiodice**

### CODE REVIEW

File: CMSIS/Core/Include/cmsis_compiler.h [hunk 1]
```diff
@@ -278,6 +278,20 @@
   #error Unknown compiler.
 #endif
 
+/*
+ * @brief Function to extract two 8-bit values from an operand (op1) rotated by 8/16/24 bits (rotate) and sign-extend them to 16 bits each
+ * @param[in] op1     input operand
+ * @param[in] rotate  number of bits (allowed 8/16/24) to rotate the input operand
+ * @return            the 8-bit values sign-extended to 16-bit values
+ */
+__STATIC_FORCEINLINE uint32_t __SXTB16_RORn(uint32_t op1, uint32_t rotate)
```
Comment (JonatanAntoni) [inline CMSIS/Core/Include/cmsis_compiler.h]: We don't have any function definitions in this selector-header, yet. Can we move this function next to the relating ones into the specific compiler headers, please?

**Comment by JonatanAntoni** [comment]
Please remember updating the file change date/version of touched files as well.

File: CMSIS/Core/Include/cmsis_compiler.h (hunk 1 above)
Comment (gmiodice) [reply CMSIS/Core/Include/cmsis_compiler.h]: Hi @JonatanAntoni ,

sorry for my late reply but I was traveling last week and I did not check the comments on github. Does this mean that I need to move this function to cmsis_gcc.h ?

Comment (JonatanAntoni) [reply CMSIS/Core/Include/cmsis_compiler.h]: Yes, and probably add it to the other compiler headers as well.

Comment (gmiodice) [reply CMSIS/Core/Include/cmsis_compiler.h]: Hi @JonatanAntoni, I have added the support in the compilers as suggested

File: CMSIS/Core/Include/cmsis_iccarm.h [hunk 2]
```diff
@@ -961,4 +961,6 @@ __packed struct  __iar_u32 { uint32_t v; };
 #pragma diag_default=Pe940
 #pragma diag_default=Pe177
 
+  #define __SXTB16_RORn __SXTB16(__ROR(op1, rotate))
```
Comment (JonatanAntoni) [inline CMSIS/Core/Include/cmsis_iccarm.h]: @TTornblom, can you please cross check that this macro definition works for IAR compiler?

File: CMSIS/Core/Include/cmsis_armcc.h [hunk 3]
```diff
@@ -875,6 +875,10 @@ __attribute__((always_inline)) __STATIC_INLINE uint32_t __USAT(int32_t val, uint
 #define __SMMLA(ARG1,ARG2,ARG3)          ( (int32_t)((((int64_t)(ARG1) * (ARG2)) + \
                                                       ((int64_t)(ARG3) << 32U)     ) >> 32U))
 
+#define __SXTB16_RORn                    __SXTB16(__ROR(op1, rotate))
```
Comment (JonatanAntoni) [inline CMSIS/Core/Include/cmsis_armcc.h]: Did you miss the arguments to this function-like macro?
Shouldn't it be `#define __SXTB16_RORn(op1, rotate)` instead?

File: CMSIS/Core/Include/cmsis_armcc.h [hunk 4]
```diff
@@ -2,7 +2,7 @@
  * @file     cmsis_armcc.h
  * @brief    CMSIS compiler ARMCC (Arm Compiler 5) header file
  * @version  V5.1.1
- * @date     30. July 2019
+ * @date     28. January 2020
```
Comment (JonatanAntoni) [inline CMSIS/Core/Include/cmsis_armcc.h]: I'd like to ask you to increase the minor version of all touched files as well, e.g. `5.2.0` in this case.

File: CMSIS/Core/Include/cmsis_armcc.h (hunk 3 above)
Comment (gmiodice) [reply CMSIS/Core/Include/cmsis_armcc.h]: My bad. I forgot to copy the correct definition

File: CMSIS/Core/Include/cmsis_armclang.h [hunk 5]
```diff
@@ -1,8 +1,8 @@
 /**************************************************************************//**
  * @file     cmsis_armclang.h
  * @brief    CMSIS compiler armclang (Arm Compiler 6) header file
- * @version  V5.2.1
- * @date     30. July 2019
+ * @version  V5.3.1
```
Comment (JonatanAntoni) [inline CMSIS/Core/Include/cmsis_armclang.h]: I guess it should be 5.3.0

**Comment by JonatanAntoni** [comment]
Looks pretty good, now. Would you mind to change the file versions to 5.3.0 instead of 5.3.1 (which is really cosmetic)? I am happy to accept the PR, now.

Of course, we need to check further things, like documentation, tests and pack description (PDSC). Any thought on these?

File: CMSIS/Core/Include/cmsis_compiler.h [hunk 6]
```diff
@@ -278,6 +278,5 @@
   #error Unknown compiler.
 #endif
 
-
```
Comment (JonatanAntoni) [inline CMSIS/Core/Include/cmsis_compiler.h]: Would you mind to drop this whitespace change?

File: CMSIS/Core/Include/cmsis_iccarm.h [hunk 7]
```diff
@@ -961,4 +961,6 @@ __packed struct  __iar_u32 { uint32_t v; };
 #pragma diag_default=Pe940
 #pragma diag_default=Pe177
 
+#define __SXTB16_RORn(ARG1, ARG2) __SXTB16(__ROR(ARG1, ARG2))
```
Comment (JonatanAntoni) [inline CMSIS/Core/Include/cmsis_iccarm.h]: @TTornblom, may I ask you to approve this enhancement to your compiler header, please?

- [ ] Approved by IAR @TTornblom

**Pull request merged**